aiida/orm/nodes/process/calculation/calcjob.py

```diff
@@ -242,11 +244,26 @@ def set_state(self, state: CalcJobState) -> None:
 
         :raise: ValueError if state is invalid
         """
+        from aiida.manage.configuration import get_config_option
+
         if not isinstance(state, CalcJobState):
             raise ValueError(f'{state} is not a valid CalcJobState')
 
         self.set_attribute(self.CALC_JOB_STATE_KEY, state.value)
 
+        if get_config_option('calcjob_state_tracking'):
+            # it is used like this in a method below, any difference with what I am doing?
+            #timestamp = timezone.datetime_to_isoformat(timezone.now())
```

Comment:
I think this above might be better as it will store the local timezone in the log file, making it easier to interpret for local debugging (but I'm not sure, I didn't check - even if the content is the same. 

More generally, I think timezone is our library, designed exactly to avoid to have to remember to to do all the UTC timezone replacement; in any case, I'd try to generate timestamps always in the same way throughout the code, so that if we have to fix the way we generate them, it's easy to fix all occurrences?

Reply:
> I think this above might be better as it will store the local timezone in the log file, making it easier to interpret for local debugging (but I'm not sure, I didn't check - even if the content is the same.

Not sure what you mean, the call to `datetime_to_isoformat` logs stuff in a file automatically? For what is worth I get the exact same string with both options (`2021-10-01T13:16:38.792373+00:00`).

> More generally, I think timezone is our library, designed exactly to avoid to have to remember to to do all the UTC timezone replacement; in any case, I'd try to generate timestamps always in the same way throughout the code, so that if we have to fix the way we generate them, it's easy to fix all occurrences?

Yeah, I guess this is a good point. I think is also better if we don't use directly the timezone library but wrap it instead (although if this is the case I would prefer that we implement something like `get_iso_now` instead of having to call `timezone.datetime_to_isoformat` with the result from `timezone.now`, but I guess that is a different story).

Reply:
For the first part - no, I didn't mean on a file sorry, I meant that the string generated could be with the local timezone (i.e. have +02:00 at the end). However probably as you say the implementation internally is the same.